test/tests/searchTest.js

```diff
@@ -83,4 +83,58 @@ describe("Zotero.Search", function() {
 			assert.propertyVal(conditions[0], 'value', 'foo')
 		});
 	});
+
+	describe("#search()", function () {
+		let win;
+	
+		before(function () {
+			// Hidden browser, which requires a browser window, needed for charset detection
+			// (until we figure out a better way)
+			if (!Zotero.isStandalone) {
+				return loadBrowserWindow().then(window => win = window);
+			}
+		});
+
+		after(function () {
+			if (win) {
+				win.close();
+			}
+		});
+	
+		it('should return results with fulltext conditions', function* () {
+			let id = yield importFromTestFile("search_foo.html");
+			let s = new Zotero.Search();
+			s.addCondition('fulltextWord', 'contains', 'foo');
+			let results = yield s.search();
+			assert.equal(results[0], id);
+		});
+		
+		it('should not return failed matches with fulltext conditions', function* () {
+			let id = yield importFromTestFile("search_foo.html");
```

Comment:
since these are not getting cleaned up after each test, we should probably just import these files in the `before` handler once. Otherwise, this may get complicated. Also, I think you should reset the DB before running all the search tests, since they may start getting affected by other items that are imported in previous tests.